### PR TITLE
Expound on field query param for the data json format and geojson format

### LINKED ISSUE #2052: Add documentation on the `field` query param for the data viewset
### Requested Documentation Change

Currently, the documentation is a bit unclear on how the `fields` query works and where the query parameter is usable. It would be nice to add documentation on how the query parameter can be used and the type of responses it returns

### PR DESCRIPTION
**Pull request by WinnyTroy**
### Changes / Features implemented
- Updated documentation

Closes #2052

### CODE REVIEW

File: docs/data.rst [hunk 1]
```diff
@@ -287,9 +287,10 @@ Sample response with link header
       ]
 
 
-Sort submitted data of a specific form using existing fields
--------------------------------------------------------------
-Provides a sorted list of json submitted data for a specific form by specifing the order in which the query returns matching data. Use the `sort` parameter to filter the list of submissions.The sort parameter has field and value pairs.
+Sort form data against a column(field)
+----------------------------------------
+Returns submitted data for a specific form in a sorted order. Use the `sort` parameter to filter the list of submissions.
+The sort parameter is takes a field(form column) as the key and select value pairs as demonstrated below.
```
Comment (DavisRayM) [inline docs/data.rst:293]: I think the previous version was clear enough... I don't see the need to change it

File: docs/data.rst [hunk 2]
```diff
@@ -376,6 +377,58 @@ Response
         "subscriberid": "639027...60317"
     }
 
+Fetch data on select columns for a given form
+---------------------------------------------------
+
+Queries select column data specified by the user. Use the ``fields`` parameter to specify the column data that should be returned.
```
Comment (DavisRayM) [inline docs/data.rst:383]:
```suggestion
Returns a list of the selected columns from the submitted data. Use the ``fields`` parameter....
```

**Review by DavisRayM — COMMENTED** [review]
Have a few comments

**Review by DavisRayM — APPROVED** [review]
LGTM

**Review by denniswambua — APPROVED** [review]

**Pull request merged**